src/tests/simple_obj.c

```diff
@@ -0,0 +1,782 @@
+		iod.iod_nr	= 1;
+		iod.iod_size	= 1; /** record size (1 byte array here) */
+		recx.rx_nr	= BUFLEN; /** extent size */
+		recx.rx_idx	= rank * BUFLEN; /** extent offset */
+		iod.iod_recxs	= &recx;
+		iod.iod_type	= DAOS_IOD_ARRAY; /** the value type of the akey */
+
+		/*
+		 * Update a dkey. In this case we have 1 akey under this dkey,
+		 * hence 1 iod and 1 sgl. for multiple akey access, this
+		 * function is used with an array of iods and sgls and the
+		 * number of akeys passed as the nr (5th argument to this
+		 * function.
+		 */
+		rc = daos_obj_update(oh, DAOS_TX_NONE, 0, &dkey, 1, &iod, &sgl,
+				     NULL);
+		ASSERT(rc == 0, "object update failed with %d", rc);
+	}
+
+	for (i = 0; i < KEYS; i++) {
+		d_sg_list_t	sgl;
+		d_iov_t		sg_iov;
+		daos_iod_t	iod;
+		daos_recx_t	recx;
+
+		/** init dkey */
+		sprintf(dkey_str, "dkey_%d", i);
+		d_iov_set(&dkey, dkey_str, strlen(dkey_str));
+
+		/** init scatter/gather */
+		d_iov_set(&sg_iov, rbuf, BUFLEN);
+		sgl.sg_nr		= 1;
+		sgl.sg_nr_out		= 0;
+		sgl.sg_iovs		= &sg_iov;
+
+		/** init I/O descriptor */
+		d_iov_set(&iod.iod_name, "akey", strlen("akey")); /** akey */
+		iod.iod_nr	= 1; /** number of extents in recx array */
+		iod.iod_size	= 1; /** record size (1 byte array here) */
+		recx.rx_nr	= BUFLEN; /** extent size */
+		recx.rx_idx	= rank * BUFLEN; /** extent offset */
+		iod.iod_recxs	= &recx;
+		iod.iod_type	= DAOS_IOD_ARRAY; /** the value type of the akey */
+
+		/** fetch a dkey */
+		rc = daos_obj_fetch(oh, DAOS_TX_NONE, 0, &dkey, 1, &iod, &sgl,
+				    NULL, NULL);
+		ASSERT(rc == 0, "object update failed with %d", rc);
+
+		if (memcmp(buf, rbuf, BUFLEN) != 0)
+			ASSERT(0, "Data verification");
+		memset(rbuf, 0, BUFLEN);
+	}
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	/** list all dkeys */
+	enumerate_key(oh, &total_nr, NULL, OBJ_DKEY);
+	ASSERT(total_nr == KEYS, "key enumeration failed");
+
+	MPI_Barrier(MPI_COMM_WORLD);
+	if (rank == 0) {
+		/** punch/remove 1 akey */
+		sprintf(dkey_str, "dkey_%d", 2);
+		d_iov_set(&dkey, dkey_str, strlen(dkey_str));
+		rc = daos_obj_punch_dkeys(oh, DAOS_TX_NONE, 0, 1, &dkey, NULL);
+		ASSERT(rc == 0, "object punch failed with %d", rc);
+	}
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	/** list all dkeys again (should have 1 less) */
+	enumerate_key(oh, &total_nr, NULL, OBJ_DKEY);
+	ASSERT(total_nr == KEYS - 1, "key enumeration failed");
+
+	daos_obj_close(oh, NULL);
+
+	MPI_Barrier(MPI_COMM_WORLD);
+	if (rank == 0)
+		printf("SUCCESS\n");
+}
+
+void
+example_daos_key_sv()
+{
+	daos_handle_t	oh;
+	char		buf[BUFLEN], rbuf[BUFLEN];
+	daos_obj_id_t	oid;
+	d_iov_t		dkey;
+	int		total_nr = 0;
+	char		dkey_str[10];
+	char		akey_str[10];
+	int		i, rc;
+
+	MPI_Barrier(MPI_COMM_WORLD);
+	if (rank == 0)
+		printf("Example of DAOS Key Single Value type:\n");
+
+	/*
+	 * Most of this example is the same as the key_array one, except the
+	 * type of the value under tha akey will be a single value of size
```

Comment:
11:19:09  src/tests/simple_obj.c:355: tha ==> than, that, the

This is what is causing checkpatch to fail

Reply:
There is an open bug on this but it happens when the suggestion is more than one word